libsolidity/lsp/LanguageServer.cpp

```diff
@@ -347,6 +385,13 @@ void LanguageServer::handleInitialize(MessageID _id, Json::Value const& _args)
 	m_client.reply(_id, move(replyArgs));
 }
 
+void LanguageServer::handleInitialized(MessageID, Json::Value const&)
+{
+	lspDebug(fmt::format("handle initialized notification: {}", m_analyzeAllFilesFromProject ? "analyze-all-files-in-project" : "analyze-files-on-demand"));
```

Comment:
Actually, maybe this setting would be better of as an enum? It would be easier to add more strategies later and it would be more readable. For example:
```
enum FileLoadStrategy
    PreloadProjectDir
    OnImport
```

Reply:
I disagree. At least from my point of view we either include the full project for analysis, or we don't. I'd prefer not to overengineer this area and just stick to what other LSPs are doing, too (which is analyzing all project files, but for testing purposes in Soliidty's LSP, it's "convenient" to only operate on opened & imported files.